### PR TITLE
[IMPROVE] E2E Encryption room enabling and disabling

### PR DESCRIPTION
**Pull request by pierre-lehnen-rc**

### CODE REVIEW

**Review by sampaiodiego — APPROVED** [review]

File: app/e2e/client/tabbar.ts [hunk 1]
```diff
@@ -5,10 +5,11 @@ import { addAction } from '../../../client/views/room/lib/Toolbox';
 import { useSetting } from '../../../client/contexts/SettingsContext';
 import { usePermission } from '../../../client/contexts/AuthorizationContext';
 import { useMethod } from '../../../client/contexts/ServerContext';
+import { e2e } from './rocketchat.e2e';
 
 addAction('e2e', ({ room }) => {
 	const e2eEnabled = useSetting('E2E_Enable');
-	const hasPermission = usePermission('edit-room', room._id);
+	const hasPermission = usePermission('edit-room', room._id) && usePermission('toggle-room-e2e-encryption', room._id) && (e2e.isReady() || room.encrypted);
```
Comment (sampaiodiego) [inline app/e2e/client/tabbar.ts:12]: I just found out this is actually breaking for DMs.. since there is no owner on a DM, no one has this permission

**Review by sampaiodiego — CHANGES_REQUESTED** [review]

File: app/e2e/client/tabbar.ts (hunk 1 above)
Comment (pierre-lehnen-rc) [reply app/e2e/client/tabbar.ts:12]: A wise man once said: "If nobody is the owner, everybody is an owner"

should I let anybody toggle it on DMs? (even multi-party DMs)

Comment (sampaiodiego) [reply app/e2e/client/tabbar.ts:12]: yes. everybody in a DM (even mutl-party) should be able to toggle it. not sure sure if we should make them all owners or have an special condition

**Comment by pierre-lehnen-rc** [comment]
Those changes were added to #20201

**Pull request closed without merging**